### PR TITLE
Added reinforcement learning example

### PR DESCRIPTION
**Pull request by jspieler**
Hi, I thought it would be nice to have a reinforcement learning example, so I implemented the Deep Deterministic Policy Gradient algorithm using Tinygrad. It seems to be slower compared to Pytorch and Tensorflow (I guess it's the forward propagations of the target networks in the learning step). Maybe someone has an idea how to speed it up.

### CODE REVIEW

**Comment by geohot** [comment]
Run with GPU=1 to use the GPU, and DEBUG=2 to see what is slow

**Comment by jspieler** [comment]
I guess it's this one
`**CL**  12134 re_S400_300                  args     3  kernels [300, 400]         None         OPs    15.4M/  29.36G  mem  0.14 GB tm    692.22us/   547.60ms (   22.19 GFLOPS)`

and maybe this one could also be faster
`**CL**  12158 ew_S300_400_N25              args    28  kernels [400, 300]         None         OPs     9.4M/  29.39G  mem  0.14 GB tm    256.93us/   548.26ms (   36.43 GFLOPS)`
